src/lib/y2storage/volume_specification_builder.rb

```diff
@@ -0,0 +1,227 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+require "y2storage/filesystems/type"
+require "y2storage/partition_id"
+require "y2storage/proposal_settings"
+
+module Y2Storage
+  # This class is able to provide a volume specification for a given mount point.
+  #
+  # If no specification exists for a given mount point, it will try to offer
+  # a fallback.
+  #
+  # @example Volume specification for /boot
+  #   builder = VolumeSpecificationBuilder.new
+  #   builder.for("/boot")
+  #
+  # @example Non-existent volume specification
+  #   builder = VolumeSpecificationBuilder.new
+  #   builder.for("/some/mount/point") #=> nil
+  class VolumeSpecificationBuilder
+    attr_reader :proposal_settings
+
+    # Constructor
+    #
+    # @param proposal_settings [ProposalSettings] Proposal settings
+    def initialize(proposal_settings = nil)
+      @proposal_settings = proposal_settings || Y2Storage::ProposalSettings.new_for_current_product
+    end
+
+    # Return a volume specification for a given mount point
+    #
+    # It will use the volume specification found within the list of volumes
+    # in the proposal settings. If it is not found, it will try to propose
+    # a fallback.
+    #
+    # @param mount_point [String] Volume mount point
+    # @return [VolumeSpecification] Volume specification;, nil if the
+    #   specification was not found and no fallback could be proposed.
+    def for(mount_point)
+      proposal_spec(mount_point) || fallback_spec(mount_point)
+    end
+
+  private
+
+    # Return a volume spec from proposal settings for the given mount point
+    #
+    # @param mount_point [String] Volume mount point
+    # @return [VolumeSpecification,nil] Volume specification if found; otherwise,
+    #   it returns nil.
+    def proposal_spec(mount_point)
+      case proposal_settings.format
+      when :ng
+        ng_proposal_spec(mount_point)
+      when :legacy
```

Comment:
Question: is it necessary to continue supporting legacy format?

Reply:
That's a good question. And, now that you have mentioned, I would say that the legacy format is deprecated. So I will happily remove those bits :)